### PR TITLE
[ci]:Improve Kafka integration test self-sufficiency

### LINKED ISSUE #4982: [ci] Make Kafka integration test more self-siufficient
The test script `scripts/kafka-integration-test.sh` is currently very simple, it could use some improvements:
  * Instead of starting Kafka in the CI it could be started from the script when an extra `-k` parameter is passed. This makes it easier to run the test locally. Example of starting Kafka manually in #4981
  * The CI workflow `.github/workflows/ci-kafka.yml` starts both Kafka and ZK. Is ZK actually needed with the latest versions of Kafka?
  * The script has an infinite `while true` loop waiting for Kafka to start - if should instead use a loop counter and be time bound, say no more than 3min, with checks every 5sec or so

### PR DESCRIPTION
**Pull request by RipulHandoo**
## Which problem is this PR solving?
Resolves #4982 

## Description of the changes
- Introduced a -k parameter to the kafka-integration-test.sh script.
- Modified the script to wait for Kafka availability before proceeding.

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [ ] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: scripts/kafka-integration-test.sh [hunk 1]
```diff
@@ -3,9 +3,49 @@
 set -e
 
 export STORAGE=kafka
-while true; do
+
+# Function to start Kafka
+start_kafka() {
+    echo "Starting Kafka..."
+    
+    docker run --name kafka \
```
Comment (yurishkuro) [inline scripts/kafka-integration-test.sh:11]: this looks like a blocking run, don't you need to detach it? How did you test your changes?

File: scripts/kafka-integration-test.sh [hunk 2]
```diff
@@ -3,9 +3,49 @@
...
+
+# Set the timeout in seconds
+timeout=180
+# Set the interval between checks in seconds
+interval=5
+
+# Calculate the end time
+end_time=$((SECONDS + timeout))
+
+while [ $SECONDS -lt $end_time ]; do
```
Comment (yurishkuro) [inline scripts/kafka-integration-test.sh:38]: please run shellcheck linter, I think it will fail this

Comment (RipulHandoo) [reply scripts/kafka-integration-test.sh:38]: How can i test these changes locally?

Comment (yurishkuro) [reply scripts/kafka-integration-test.sh:38]: by running the script - that was the whole point of the ticket, to enable local integration test

File: scripts/kafka-integration-test.sh (hunk 1 above)
Comment (RipulHandoo) [reply scripts/kafka-integration-test.sh:11]: Here i am have added `-d` tag so the docker container will run in the background
```
docker run --name kafka -d\
--network jaeger \
    -p 9092:9092 \
    -e KAFKA_CFG_NODE_ID=0 \
    -e KAFKA_CFG_PROCESS_ROLES=controller,broker \
    -e KAFKA_CFG_CONTROLLER_QUORUM_VOTERS=0@kafka:9093 \
    -e KAFKA_CFG_LISTENERS=PLAINTEXT://:9092,CONTROLLER://:9093 \
    -e KAFKA_CFG_ADVERTISED_LISTENERS=PLAINTEXT://localhost:9092 \
    -e KAFKA_CFG_LISTENER_SECURITY_PROTOCOL_MAP=CONTROLLER:PLAINTEXT,PLAINTEXT:PLAINTEXT \
    -e KAFKA_CFG_CONTROLLER_LISTENER_NAMES=CONTROLLER \
    -e KAFKA_CFG_INTER_BROKER_LISTENER_NAME=PLAINTEXT \
    bitnami/kafka:3.6
```

Comment (yurishkuro) [reply scripts/kafka-integration-test.sh:11]: `-d` is not in the PR

Comment (RipulHandoo) [reply scripts/kafka-integration-test.sh:11]: Yes, i'll add it right away. Just waiting for confirmation.

Comment (RipulHandoo) [reply scripts/kafka-integration-test.sh]: Getting an error while running `bash scripts/kafka-integration-test.yml`

image

Comment (RipulHandoo) [reply scripts/kafka-integration-test.sh]: Getting an error. I think that error is related to a missing dependency.
image

Comment (RipulHandoo) [reply scripts/kafka-integration-test.sh]: Acknowledged. I overlooked creating the network in the previous PR, resulting in test failures. The PR has been updated. Please review when you can. Thank you

File: scripts/kafka-integration-test.sh [hunk 3]
```diff
@@ -3,9 +3,74 @@
...
+    docker network inspect jaeger > /dev/null 2>&1
+}
+
+# Function to check if Kafka container is running
+kafka_container_running() {
+    docker inspect -f '{{.State.Running}}' kafka 2>/dev/null || echo "false"
+}
+
+# Function to create the Jaeger network if it doesn't exist
+create_jaeger_network() {
```
[the remark this answers is not in the document]
Comment (RipulHandoo) [reply scripts/kafka-integration-test.sh:18]: Without this` kafka-integration-test` are failing.
With Error message 
image

[the remark this answers is not in the document]
Comment (RipulHandoo) [reply scripts/kafka-integration-test.sh]: @james-ryans I've removed the `jaeger network.` PTAL.

**Comment by yurishkuro** [comment]
This is what DCO bot says:

> Commit sha: [c5cde85](https://github.com/jaegertracing/jaeger/pull/4983/commits/c5cde85597270f3cbfefe940c66a7e20212424b1), Author: root, Committer: root; root@LAPTOP-KAK9J9IH is not a valid email address.

**Comment by yurishkuro** [comment]
See CONTRIBUTING_GUIDELINES.md on how to squash commits reliably

**Pull request closed without merging**